### PR TITLE
Add dynamic-tcp-client Sample

### PR DESCRIPTION
**Pull request by garyrussell**
Utilizing the new DSL dynamic flow registration.

### CODE REVIEW

File: advanced/dynamic-tcp-client/src/main/java/org/springframework/integration/samples/dynamictcp/DynamicTcpClientApplication.java [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+import org.springframework.integration.ip.tcp.connection.TcpNetServerConnectionFactory;
+import org.springframework.integration.router.AbstractMessageRouter;
+import org.springframework.messaging.Message;
+import org.springframework.messaging.MessageChannel;
+import org.springframework.messaging.handler.annotation.Header;
+import org.springframework.util.Assert;
+
+@SpringBootApplication
+@IntegrationComponentScan
+@EnableMessageHistory
```
Comment (artembilan) [inline advanced/dynamic-tcp-client/src/main/java/org/springframework/integration/samples/dynamictcp/DynamicTcpClientApplication.java]: Well, I think I should come back to my Spring Boot `IntegrationProperties` work and utilize this feature as an auto-config property.
I think we can do the same for the `@IntegrationComponentScan` as well...

Right, different storty, but I'd like to start a discussion on the matter.

File: advanced/dynamic-tcp-client/src/main/java/org/springframework/integration/samples/dynamictcp/DynamicTcpClientApplication.java [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+
+			TcpNetClientConnectionFactory cf = new TcpNetClientConnectionFactory(host, port);
+			this.context.getBeanFactory().registerSingleton(hostPort + ".cf", cf);
+			this.context.getBeanFactory().initializeBean(cf, hostPort+".cf");
+			cf.afterPropertiesSet();
+			TcpSendingMessageHandler handler = new TcpSendingMessageHandler();
+			handler.setConnectionFactory(cf);
+			IntegrationFlow flow = f -> f.handle(handler);
+			this.flowContext.register(hostPort + ".flow", flow);
+			MessageChannel channel = context.getBean(hostPort + ".flow.input", MessageChannel.class);
```
Comment (artembilan) [inline advanced/dynamic-tcp-client/src/main/java/org/springframework/integration/samples/dynamictcp/DynamicTcpClientApplication.java]: I think we can avoid `ctx.getBean()` here with the `this.flowContext.messagingTemplateFor(hostPort + ".flow").getDefaultDestination()`. That's why I suggest one more option-method in a refinement of the `IntegrationFlowContext`: https://github.com/spring-projects/spring-integration-java-dsl/issues/125

WDYT?

Comment (garyrussell) [inline advanced/dynamic-tcp-client/src/main/java/org/springframework/integration/samples/dynamictcp/DynamicTcpClientApplication.java]: Yes, thanks, didn't notice that method and yes, I agree with the other discussion.

**Comment by artembilan** [comment]
Merged as https://github.com/spring-projects/spring-integration-samples/commit/2e103f6f34f65824cabee44e63f274ed5bf2a6c9

**Pull request closed without merging**